### ISSUE TITLE
Adminer should default to relevant DB

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#360]
Currently Adminer (replaing phpMyAdmin & phpPgAdmin for v14.0) is not preconfigured at all. This is not really an issue but perhaps we could make the default a little nicer.

If host is left blank then it defaults to localhost although perhaps this could be more explicit?

For appliances with MySQL the default is 'MySQL' (slectable from a dropdown) so not too much of an issue.

But for PostgreSQL appliances 'PostgreSQL' needs to be slected from the dropdown. It would be nice if this was configured by default.

**Comment by JedMeister** [comment]
As per @DocCyblade comment on https://github.com/turnkeylinux/tracker/issues/387#issuecomment-110968038 I'm assigning htis issue to him! :smile:

**Comment by DocCyblade** [comment]
TODO Note: create code to uncomment un-needed includes for the database drivers. Also look into adding an include or edit index.php to inject 127.0.0.1 or localhost so that it displays by default.

**Comment by DocCyblade** [comment]
@JedMeister - Thoughts on how to implement this. Should I create the sed statements to remove the needed ones per appliance? I know I will need to do this with postgres and mysql since they don't share lamp/lamp common. As for the lamp/lapp many other builds depend on these so anything I do to common will be on every single other appliance that depends on it. So do we want all the other apps to have only the mysql database selection for lamp and postgres for lapp? OR should we leave it up to the individual apps. I am leaning on making the change in common for lamp/lapp to keep it DRY and would make for a consistent appliance builds that depend on lamp/lapp.

I just had a thought I could add a adminer-mysql / adminer-postgres conf script in common that would augment the adminer install (remove the includes) them just reference them from both the lamp/lapp and mysql/postgres appliances. That seems even better.

Thoughts?

**Comment by JedMeister** [comment]
I agree to do it in common. I'm not totally sure of the best place/approach though. The options as I see them:
1. Use a script (i.e. sed probably) to remove the irrelevant ones
2. As per 1 but use an overlay file instead
3. Set the relevant DB to be the default (and leave others as is)

Originally I though that 3 would be the go. The advantage of that is that LAMP and MySQL appliance(s) won't need anything (it defaults to MySQL already). But 1 or 2 would be neater solutions!

One other thing that I recall is PostgrSQL also requires you to fill in the server field (localhost works fine). We should pre-populate that too...

Bottom line is that I think that so long as you can provide a good rationale for why you decided to go a particular direction then it's good. Decision making is often more about the process than the answer. An answer is 'right' if you used a reasonable process to get to it! :smile:

**Comment by DocCyblade** [comment]
Well, I was having issues building on my TKLDev VM. Turns out was a proxy/cache issue. [See here for details](https://www.turnkeylinux.org/forum/support/20150701/tkldev-builds-very-slow-build#comment-23419) on that fun adventure.

Now I have my builds building again, I can work on this!

**Comment by DocCyblade** [comment]
Hacking away at it tonight. I am double checking to make sure that commenting out the includes for the database driver does not affect adminer functionality for the other database drivers.

**Comment by DocCyblade** [comment]
Pulled this from another post:

## Adminer Database Selection FYI ##

Just an FYI for adminer, it seems to disable or make a default database type (disable all but the one you want) Just comment out the included database drivers. The below snippet will show only postgres

Below snippet from file
**/usr/share/adminer/adminer/include/bootstrap.inc.php**
```php
include "../adminer/include/lang.inc.php";
include "../adminer/lang/$LANG.inc.php";
// include "../adminer/include/pdo.inc.php";
// include "../adminer/drivers/sqlite.inc.php";
include "../adminer/drivers/pgsql.inc.php";
// include "../adminer/drivers/oracle.inc.php";
// include "../adminer/drivers/mssql.inc.php";
// include "../adminer/drivers/mysql.inc.php"; 
```

**Comment by DocCyblade** [comment]
Post another post:

Researching on how to extent adminer found this. Maybe able to use this as an include to customize adminer to our liking.

http://www.adminer.org/en/extension/

**Comment by DocCyblade** [comment]
In my hacking away at Adminer PHP script, I found this line and commented out and added my own.

```php
// define("SERVER", $_GET[DRIVER]); // read from pgsql=localhost
define("SERVER", "localhost");
```

Seems to make SERVER always localhost

Need to test it!!

**Comment by JedMeister** [comment]
>In my hacking away at Adminer PHP script, I found this line and commented out and added my own.
````
// define("SERVER", $_GET[DRIVER]); // read from pgsql=localhost
define("SERVER", "localhost");
````
Seems to make SERVER always localhost

>Need to test it!!

Awesome work Ken! Sounds like you are all over it! :+1: 

So you thinking that we should tweak that with sed? That's probably what I'd do but I'll leave that one with you seeing as you're doing the work!

**Comment by JedMeister** [comment]
FWIW I had a bit of a play/test and you are totally on the money @DocCyblade. I tested your line above and it works a treat (i.e. I edited `include/bootstrap.inc.php`).

Also I had a bit of a fiddle with commenting out the lines that include the DB drivers (also in `include/bootstrap.inc.php`). If you comment out DBs that you don't want it removes them as options in the drop down. E.g. for MySQL
````
include "../adminer/include/lang.inc.php";
include "../adminer/lang/$LANG.inc.php";
include "../adminer/include/pdo.inc.php";
//include "../adminer/drivers/sqlite.inc.php";
//include "../adminer/drivers/pgsql.inc.php";
//include "../adminer/drivers/oracle.inc.php";
//include "../adminer/drivers/mssql.inc.php";
include "../adminer/drivers/mysql.inc.php"; // must be included as last driver
````
I tested PostgreSQL too and it looks like it works. Although I didn't do a proper live test (as I was using an appliance that didn't have it installed).

Also. It doesn't quite fit here but thought I'd post anyway, I got rid of the new version download link. It's in `include/adminer.inc.php`. This is the line to remove:
````
<a href="http://www.adminer.org/#download" id="version"><?php echo (version_compare($VERSION, $_COOKIE["adminer_version"]) < 0 ? h($_COOKIE["adminer_version"]) : ""); ?></a>
````

**Comment by DocCyblade** [comment]
Awesome, we should add that sed command to patch adminer. Maybe in common in a conf script so we can use it everywhere? 

I'll be working on the sed statements in common on Lapp and lamp I could throw that in as well.

**Comment by DocCyblade** [comment]
Ok so below are the sed statements that will do the commenting of the lines.

```bash
# For LAMP / MySQL
sed -i '/^include "..\/adminer\/drivers\/sqlite.inc.php"/ s/include/\/\/ include/' fileNameHere
sed -i '/^include "..\/adminer\/drivers\/pgsql.inc.php"/ s/include/\/\/ include/' fileNameHere
sed -i '/^include "..\/adminer\/drivers\/oracle.inc.php"/ s/include/\/\/ include/' fileNameHere
sed -i '/^include "..\/adminer\/drivers\/mssql.inc.php"/ s/include/\/\/ include/' fileNameHere


# For LAPP/PostgreSQL
sed -i '/^include "..\/adminer\/drivers\/sqlite.inc.php"/ s/include/\/\/ include/' fileNameHere
sed -i '/^include "..\/adminer\/drivers\/oracle.inc.php"/ s/include/\/\/ include/' fileNameHere
sed -i '/^include "..\/adminer\/drivers\/mssql.inc.php"/ s/include/\/\/ include/' fileNameHere
sed -i '/^include "..\/adminer\/drivers\/mysql.inc.php"/ s/include/\/\/ include/' fileNameHere
```

I am have a heck of a time with the local host settings. 

I am looking for this `define("SERVER", $_GET[DRIVER])`

Replace `$_GET[DRIVER]` with `"localhost"`

I thought I had it with:
`sed -i '/^define("SERVER", $_GET[DRIVER])/ s/define("SERVER", $_GET[DRIVER]/"localhost"/' fileNameHere`

But I think I need to escape some of the special characters. Just out of brain power for the night.

**Comment by JedMeister** [comment]
Yep you need to escape the `$` or otherwise it will be expanded as a variable. You also need to escape the brackets (IIRC both square and round).

The way I like to test sed statements is like this:
````
cat FILENAME | grep OLD-PATTERN | sed 's|OLD-PATTERN|NEW_PATTERN|' 
````
Then once it works exactly as I intend I use:
````
sed -i 's|OLD-PATTERN|NEW_PATTERN|' FILENAME
````

**Comment by DocCyblade** [comment]
Your way of testing.... brilliant! We need to put together a best practices or this is how we do it blog post. 

Made quick work of things! I should have a few pull requests tonight

**Comment by DocCyblade** [comment]
Testing common updates now, so far so good. Lamp build good, building postgres now

**Comment by DocCyblade** [comment]
both lamp and lapp builds with changes in hand. Changes the default database, sets localhost as default server and removes the update link. 

Opened pull request https://github.com/turnkeylinux/common/pull/39

**Comment by l-arnold** [comment]
Nice Work Ken!

**Comment by DocCyblade** [comment]
Since this is merged, this can be closed, mark another one off my list!

**Comment by JedMeister** [comment]
Nice one mate! :+1:
